augur/util_support/date_disambiguator.py

```diff
@@ -60,7 +60,32 @@ def range(self):
             resolve_uncertain_int(self.uncertain_date_components["m"], "max"),
             resolve_uncertain_int(self.uncertain_date_components["d"], "max"),
         )
-        max_date = min(max_date, datetime.date.today())
+
+        # Set default bounds.
+        lower_bound = None
+        upper_bound = datetime.date.today()
+
+        # TODO: Allow more granularity than year alone.
+        if self.min_max_year:
+            lower_bound = datetime.date(int(self.min_max_year[0]), 1, 1)
+
+            if len(self.min_max_year) == 2:
+                upper_bound = datetime.date(int(self.min_max_year[1]), 12, 31)
+
+        # Limit dates to the given bounds.
+        if lower_bound:
+            if lower_bound > max_date:
+                raise InvalidDate(self.uncertain_date, f"Not possible for date to fall within bounds '{self.min_max_year}'")
```

Comment:
Best, I think, to reveal the defaults that are being applied as their use may not be obvious otherwise.

```suggestion
                raise InvalidDate(self.uncertain_date, f"Not possible for date to fall within bounds [{lower_bound or ''}, {upper_bound or ''}]")
```

Reply:
Good idea on revealing defaults. I think I'll go with `[{lower_bound}, {upper_bound}]` for simplicity, as the output makes sense too:

```
# [{lower_bound or ''}, {upper_bound or ''}]
Invalid date '2024-01-31': Not possible for date to fall within bounds [, 2023-01-31]

# [{lower_bound}, {upper_bound}]
Invalid date '2024-01-31': Not possible for date to fall within bounds [None, 2023-01-31]
```

Reply:
Done: 525bfd86835ea4b320087fa779fbfd4a2eb77f9a

Reply:
Sure. I was going for the typical syntax for an unbounded range, but `None` is probably also just as interpretable.